src/OnlineUpdateDialogs.ycp

```diff
@@ -357,4 +359,81 @@ some of them might not work as expected.");
 	else return false;
     }
 
+    // Returns all installed products
+    list<map> installed_products()
+    {
+        list<map> products = Pkg::ResolvableProperties("", `product, "");
+
+        return (list<map>)filter(map p, products, {
+            return p["status"]:nil == `installed;
+        });
+    }
+
+    // Returns all products over the end of life support
+    list<map> eol_products()
+    {
+        integer time_now = time();
+        list<map> products = installed_products();
+
+        return (list<map>)filter(map p, products, {
```

Comment:
Can't you just use `filter(map p, installed_products, {`? Variable `products` is not used anywhere else. Maybe just nitpicking...